lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1657,8 +1656,19 @@ namespace {
     : public clang::ConstDeclVisitor<SwiftDeclConverter, Decl *>
   {
     ClangImporter::Implementation &Impl;
-    bool useSwift2Name;
     bool forwardDeclaration = false;
+    ImportNameVersion version;
+
+    /// The version that we're being asked to import for
+    ImportNameVersion getVersion() const { return version; }
+
+    /// The actual language version we're compiling for overall
+    ImportNameVersion getCorrectVersion() const { return CurrentVersion; }
+
+    /// Whether the names we're importing are from the actual language version
+    bool isCorrectSwiftName() const {
+      return getVersion() == getCorrectVersion();
```

Comment:
When in doubt, be explicit. SwiftVersionBeingCompiled vs. SwiftVersionToImport? With "isActive" as the predicate?